website_sale_product_minimal_price/controllers/main.py

```diff
@@ -19,30 +19,70 @@ def get_combination_info_minimal_price(self, product_template_ids, **kw):
         """Special route to use website logic in get_combination_info override.
         This route is called in JS by appending _website to the base route.
         """
-
         res = []
         templates = request.env["product.template"].sudo().browse(product_template_ids)
+        pricelist = request.env["website"].get_current_website().get_current_pricelist()
         for template in templates.filtered(lambda t: t.is_published):
-            cheaper_variant = template.product_variant_ids.sorted(
-                key=lambda p: p._get_combination_info_variant().get("price")
-            )[:1]
+            product_id, add_qty, has_distinct_price = template._get_cheapest_info(
+                pricelist
+            )
+            combination = template._get_combination_info(
+                product_id=product_id, add_qty=add_qty, pricelist=pricelist
+            )
             res.append(
                 {
                     "id": template.id,
-                    "price": cheaper_variant._get_combination_info_variant().get(
-                        "price"
-                    ),
-                    "distinct_prices": self._compute_has_distinct_variant_price(
-                        template
-                    ),
+                    "price": combination.get("price"),
+                    "distinct_prices": has_distinct_price,
                 }
             )
-
         return res
 
-    def _compute_has_distinct_variant_price(self, template):
-        if template.product_variant_count > 1:
-            prices = template.product_variant_ids.mapped("price")
-            if len(prices) > 1:
-                return True
-        return False
+    @http.route(
+        ["/sale/get_combination_info_pricelist_atributes"],
+        type="json",
+        auth="public",
+        website=True,
+    )
+    def get_combination_info_pricelist_atributes(self, product_id, **kwargs):
+        """Special route to use website logic in get_combination_info override.
+        This route is called in JS by appending _website to the base route.
+        """
+        product = request.env["product.product"].browse(product_id)
+        pricelist = request.env["website"].get_current_website().get_current_pricelist()
+        # Getting all min_quantity of the current product to compute the possible
+        # price scale.
+        qty_list = request.env["product.pricelist.item"].search(
+            [
+                ("|"),
```

Comment:
I'm seeing that this is incorrect syntax. It should be just:
```suggestion
                "|",
```

If it works, it's by chance.